python/pyspark/sql/types.py

```diff
@@ -71,6 +67,15 @@ def json(self):
                           separators=(',', ':'),
                           sort_keys=True)
 
+    def needSerialization(self):
```

Comment:
I guess that the rationale for this vs. implementing the identity function as a serializer is the fact that we may be dealing with nested data structures like lists and maps so we want to avoid applying the identity function to each element of the array by skipping the conversion entirely if we have an array of types that don't need to be serialized / converted.

Maybe we should explain this?